### PR TITLE
Clam 2612 Fix DatabaseCustomURL CVD prune bug, and add 'valhalla' optional database

### PR DESCRIPTION
**Pull request by micahsnyder**
- Freshclam: fix issue DatabaseCustomURL CVD prune issue

   If using DatabaseCustomURL to download a CVD that Freshclam doesn't know about, i.e. one that is not in the hardcoded standard or optional database lists in freshclam.c, Freshclam will prune the database and then re-download it.

   This change makes it so we look for URL's with ".cvd" at the end and then take those into consideration when checking which  CVD's (or CLD's) should be pruned.

   Note that I didn't change the interface to fc_prune_database_directory(). That would have been cleaner, but would've changed the public API and I want to backport this fix.

- Add 'valhalla' to Freshclam's list of optional CVD's

### CODE REVIEW

File: freshclam/freshclam.c [hunk 1]
```diff
@@ -1475,7 +1475,38 @@ fc_error_t perform_database_update(
...
+        // include the URL databases in the prune process
+        char **newDatabaseList = (char **)malloc(sizeof(char *) * (nDatabases + nUrlDatabases));
+        if (NULL == newDatabaseList) {
+            logg(LOGG_ERROR, "perform_database_update: Can't allocate memory for newDatabaseList\n");
+            status = FC_EMEM;
+            goto done;
+        }
+
+        for (i = 0; i < nDatabases; i++) {
+            newDatabaseList[i] = strdup(databaseList[i]);
```
Comment (ragusaa) [inline freshclam/freshclam.c:1489]: Should you check for NULL in case strdup fails?

File: freshclam/freshclam.c [hunk 2]
```diff
@@ -1475,7 +1475,38 @@ fc_error_t perform_database_update(
...
+        }
+        uint32_t nNewDatabases = nDatabases;
+
+        for (i = 0; i < nUrlDatabases; i++) {
+            // only append the URL databases that end with '.cvd'
+            if (strlen(urlDatabaseList[i]) > 4 && 0 == strcasecmp(urlDatabaseList[i] + strlen(urlDatabaseList[i]) - 4, ".cvd")) {
+                const char* startOfFilename = strrchr(urlDatabaseList[i], '/') + 1;
```
Comment (ragusaa) [inline freshclam/freshclam.c:1496]: Where do the urls come from?  Is it possible that they would be malformed and not contain a '/'?

**Comment by ragusaa** [comment]
Testing passed

File: freshclam/freshclam.c (hunk 2 above)
Comment (micahsnyder) [reply freshclam/freshclam.c:1496]: The URL's come from the freshclam.conf `DatabaseCustomURL` option. It would be a misconfiguration. Could happen.

**Review by ragusaa — APPROVED** [review]

**Comment by micahsnyder** [comment]
Oops. There's a windows build issue:
> `freshclam.c.obj : error LNK2019: unresolved external symbol strndup referenced in function perform_database_update`

I'll fix it.

**Pull request merged**